src/lib/authserver/ui/new_dir_inst.rb

```diff
@@ -72,64 +78,72 @@ def ok_handler
       return
     end
-    if !File.exists?(tls_ca) || !File.exists?(tls_p12)
-      Popup.Error(_('TLS certificate authority or certificate/key file does not exist.'))
+    if ! ((tls_ca == '' && tls_p12 == '') || (tls_ca != '' && tls_p12 != ''))
+      Popup.Error(_('Both TLS Certificate authority and PKCS12 must be provided, or none provided.'))
       return
     end
-    if DS389.get_instance_names.include?(instance_name)
-      Popup.Error(_('The instance name is already used.'))
+    if (tls_ca != '' && tls_p12 != '') && (!File.exists?(tls_ca) || !File.exists?(tls_p12))
+      Popup.Error(_('TLS certificate authority PEM OR certificate/key PKCS12 file does not exist.'))
       return
     end
+    # The dscreate tool has an instance name checker that is much more aware of instance
+    # rules than this ruby tool can be.
+    UI.ReplaceWidget(Id(:busy), Label(_('Preparing to install new instance, this may take a minute ...')))
 
-    UI.ReplaceWidget(Id(:busy), Label(_('Installing new instance, this may take a minute or two.')))
-    begin
-      DS389.install_pkgs
-      # Collect setup parameters into an INI file and feed it into 389 setup script
-      ok = DS389.exec_setup(DS389.gen_setup_ini(fqdn, instance_name, suffix, dm_dn, dm_pass))
-      DS389.remove_setup_ini
-      if !ok
-        Popup.Error(_('Failed to set up new instance! Log output may be found in %s') % [DS_SETUP_LOG_PATH])
-        raise
-      end
+    if !DS389.install_pkgs
+      Popup.Error(_('Error during package installation.'))
+      return
+    end
+
+    # Collect setup parameters into an INI file and feed it into 389 setup script
+    ini_content = DS389.gen_setup_ini(fqdn, instance_name, suffix, dm_pass)
+    ini_safe_content = DS389.gen_setup_ini(fqdn, instance_name, suffix, "********")
+    log.debug(ini_safe_content)
+    UI.ReplaceWidget(Id(:busy), Label(_('Installing new instance, this may take a minute ...')))
+    ok = DS389.exec_setup(ini_content)
+    # Always remove the ini file
+    DS389.remove_setup_ini
+    if !ok
+      Popup.Error(_('Failed to set up new instance! Log output may be found in ~/.y2log or /var/log/YaST/y2log'))
+      UI.ReplaceWidget(Id(:busy), Empty())
+      return
+    end
+
+    if (tls_ca != '' && tls_p12 != '')
+      UI.ReplaceWidget(Id(:busy), Label(_('Configuring instance TLS ...')))
       # Turn on TLS
       if !DS389.install_tls_in_nss(instance_name, tls_ca, tls_p12)
-        Popup.Error(_('Failed to set up new instance! Log output may be found in %s') % [DS_SETUP_LOG_PATH])
-        raise
-      end
-      ldap = LDAPClient.new('ldap://'+fqdn, dm_dn, dm_pass)
-      out, ok = ldap.modify(DS389.get_enable_tls_ldif, true)
-      DS389.append_to_log(out)
-      if !ok
-        Popup.Error(_('Failed to enable TLS! Log output may be found in %s') % [DS_SETUP_LOG_PATH])
-        raise
+        Popup.Error(_('Failed to set up new instance! Log output may be found in ~/.y2log or /var/log/YaST/y2log'))
```

Comment:
it is message for user. So I would mention only /var/log/YaST/y2log as in production this module runs as root.